spec/models/notification_spec.rb

```diff
@@ -63,7 +63,33 @@
       end
     end
 
-    context "when WhatsApp flag is off" do
+    context "when imo flag is on" do
+      before { Flipper.enable(:imo_messaging) }
+
+      it "returns sms if patient can't receive Imo notifications" do
+        expect(notification.next_communication_type).to eq("sms")
```

Comment:
Let's make an explicit ImoAuthorization with `status: :error`?

Reply:
That might be wise. Lemme think about that a bit.

Reply:
I did add this. It only applies to the invitation job.

Reply:
I misread you. That's why I said, "This seems like an odd place for the comment." You're right and I'm surprised I didn't add a test for it.

Reply:
Ok, I had to look at this again. So I did test it, just not as clearly as I could have. Got it. Sorry for the confusion. I read you as saying, "We should have an error state in ImoAuth." And I ended up adding an error state to imo auth. So now if we receive an unknown Imo error, we log it in sentry and add the ImoAuth with an error state to prevent continuous retries.